boards/shared/Inc/Io/Io_SharedFreqOnlyPwmInput.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+#include "Io_SharedFreqOnlyPwmInputConfig.h"
+
+#ifndef MAX_NUM_OF_FREQ_ONLY_PWM_INPUTS
+#define MAX_NUM_OF_FREQ_ONLY_PWM_INPUTS 1
+#endif
+
+struct FreqOnlyPwmInput;
+
+/**
+ * Allocate and initialize a PWM input using the given (hardware) timer
+ *
+ * @note The given timer must be initialized with:
+ *       - Input Capture direct mode
+ *
+ * @param htim: The handle of the timer measuring the PWM input
```

Comment:
nit; mention it's freq only
```suggestion
 * Allocate and initialize a frequency-only PWM input using the given (hardware) timer
 *
 * @note The given timer must be initialized with Input Capture Direct Mode
 * @param htim: The handle of the timer measuring the PWM input
```